python/plugins/processing/algs/qgis/Rasterize.py

```diff
@@ -0,0 +1,282 @@
+                                             'Map theme to render.'),
+                                         defaultValue=None, optional=True))
+
+        self.addParameter(
+            QgsProcessingParameterRasterLayer(self.LAYER, description=self.tr(
+                'Layer to render. Will only be used if the map theme is not '
+                'set. '
+                'If both, map theme and layer are not '
+                'set, the current map content will be rendered.'),
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterExtent(self.EXTENT, description=self.tr(
+                'The minimum extent to render. Will internally be extended to '
+                'be '
+                'a multiple of the tile sizes.')))
+        self.addParameter(
+            QgsProcessingParameterNumber(self.TILE_SIZE, self.tr('Tile size'),
+                                         defaultValue=1024))
+        self.addParameter(QgsProcessingParameterNumber(self.MAP_UNITS_PER_PIXEL,
+                                                       self.tr(
+                                                           'Map units per '
+                                                           'pixel'),
+                                                       defaultValue=100))
+
+        # We add a raster layer as output
+        self.addParameter(QgsProcessingParameterRasterDestination(
+            self.OUTPUT_LAYER,
+            self.tr(
+                'Output layer')))
+
+    def name(self):
+        # Unique (non-user visible) name of algorithm
+        return 'Rasterize'
+
+    def displayName(self):
+        # The name that the user will see in the toolbox
+        return self.tr('Convert map to raster')
+
+    def group(self):
+        return self.tr('Raster tools')
+
+    # def processAlgorithm(self, progress):
+    def processAlgorithm(self, parameters, context, feedback):
+        """Here is where the processing itself takes place."""
+
+        # The first thing to do is retrieve the values of the parameters
+        # entered by the user
+        map_theme = self.parameterAsString(parameters, self.MAP_THEME, context)
+        layer = self.parameterAsString(parameters, self.LAYER, context)
+        extent = self.parameterAsExtent(parameters, self.EXTENT,
+                                        context)
+        tile_size = self.parameterAsInt(parameters, self.TILE_SIZE, context)
+        mupp = self.parameterAsInt(parameters, self.MAP_UNITS_PER_PIXEL, context)
+
+        output_layer = self.parameterAsOutputLayer(parameters, self.OUTPUT_LAYER,
+                                                   context)
+
+        # This probably affects the whole system but it's a lot nicer
+        osgeo.gdal.UseExceptions()
+
+        tile_set = TileSet(map_theme, layer, extent, tile_size, mupp, output_layer,
+                           qgis.utils.iface.mapCanvas().mapSettings())
+        tile_set.render()
```

Comment:
Can you add `feedback` as a parameter to render and add appropriate hooks within render to check for feedback.isCanceled() and abort the render early?